### PR TITLE
Remove selector field if it's empty from manifest

### LINKED ISSUE #38880: CronJob deployment doesn't work due to `selector` field
### Describe the bug

When a CLI Quarkus app is attempted to be deployed to openshift, it fails due to selector being specified. It seems, for job the `selector` should not be set preferably as per [docs](https://kubernetes.io/docs/concepts/workloads/controllers/job/#pod-selector). When I try to disable generating the labels, i.e. in `application.properties`:
`quarkus.openshift.add-version-to-label-selectors=false
quarkus.openshift.add-name-to-label-selectors=false`
it generates empty `selector: {}` in the manifest, which also fails to get deployed. Once this line is removed from the manifest, the job is successfully deployed.

### Expected behavior

Deploy the Job's manifest successfully without manual updates.

### Actual behavior

Failure to deploy the Job's manifest

[error]: Build step io.quarkus.kubernetes.deployment.KubernetesDeployer#deploy threw an exception: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://api.sandbox-m2.ll9k.p1.openshiftapps.com:6443/apis/batch/v1/namespaces/rhn-support-rbudinsk-dev/cronjobs. Message: CronJob.batch "test" is invalid: spec.jobTemplate.spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string(nil), MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: `selector` will be auto-generated. Received status: Status(apiVersion=v1, code=422, details=StatusDetails(causes=[StatusCause(field=spec.jobTemplate.spec.selector, message=Invalid value: v1.LabelSelector{MatchLabels:map[string]string(nil), MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: `selector` will be auto-generated, reason=FieldValueInvalid, additionalProperties={})], group=batch, kind=CronJob, name=test, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=CronJob.batch "test" is invalid: spec.jobTemplate.spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string(nil), MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: `selector` will be auto-generated, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Invalid, status=Failure, additionalProperties={}).

where we can see message from kubernetes proposing solution: `selector` will be auto-generated.

### How to Reproduce?

Create a simple CLI app:

1. Create demo cli app `q create cli --extension=openshift com.thehuginn:test -P io.quarkus:quarkus-bom:999-SNAPSHOT`

2. Update `application.properties` as following:

`quarkus.kubernetes.cron-job.schedule=* * * * 0
quarkus.openshift.add-version-to-label-selectors=false
quarkus.openshift.add-name-to-label-selectors=false
quarkus.kubernetes.deployment-kind=cron-job
`

3. Log into openshift, for example sandbox via `oc login ...`

4. Run the following command `./mvnw clean package -Dquarkus.openshift.deploy=true`

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by The-Huginn**
- Resolves #38880

### CODE REVIEW

**Comment by The-Huginn** [comment]
However, I was not able to run the test locally, none from the kubernetes it test suite. Could somebody please assist me here, how to run it locally?

command ran: `mvn clean test -f integration-tests/kubernetes/quarkus-standard-way -Dtest=io.quarkus.it.kubernetes.KubernetesWithCronJobResourceTest`
Full stack-trace:

[ERROR] io.quarkus.it.kubernetes.KubernetesWithCronJobResourceTest -- Time elapsed: 1.630 s <<< ERROR!
java.util.ServiceConfigurationError: io.smallrye.config.ConfigSourceFactory: io.smallrye.config.PropertiesLocationConfigSourceFactory not a subtype
        at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:593)
        at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1244)
        at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1273)
        at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1309)
        at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1393)
        at io.smallrye.config.SmallRyeConfigBuilder.discoverSources(SmallRyeConfigBuilder.java:138)
        at io.smallrye.config.SmallRyeConfig$ConfigSources.buildSources(SmallRyeConfig.java:764)
        at io.smallrye.config.SmallRyeConfig$ConfigSources.<init>(SmallRyeConfig.java:715)
        at io.smallrye.config.SmallRyeConfig.<init>(SmallRyeConfig.java:81)
        at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:724)
        at io.quarkus.deployment.configuration.BuildTimeConfigurationReader.initConfiguration(BuildTimeConfigurationReader.java:409)
        at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:137)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:107)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:330)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createProductionApplication(AugmentActionImpl.java:175)
        at io.quarkus.test.QuarkusProdModeTest.beforeAll(QuarkusProdModeTest.java:481)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

**Comment by geoand** [comment]
The command above works for me, so I assume you have not [built](https://github.com/quarkusio/quarkus/blob/main/CONTRIBUTING.md#building-main) the entire project first.

**Review by geoand — APPROVED** [review]

**Comment by The-Huginn** [comment]
Thanks, after clean rebuild it works

**Comment by The-Huginn** [comment]
@geoand should we even at first place allow to apply the labels to cronjobs, if it doesn't seem to be recommended according to the [docs](https://kubernetes.io/docs/concepts/workloads/controllers/job/#pod-selector)?
Current fix doesn't work, as the labels are not applied at that stage anyways, so it only preservers existing labels and `version` and `name` label are applied later on to all resources. That's why all other existing tests fail, as they expect selector

**Comment by geoand** [comment]
I'll leave that to @iocanel since I have not followed the developments in k8s for a while

**Comment by iocanel** [comment]
What problem are we trying to solve here? If the presence of selectors is causing issue, I am all for fixing them. But IMHO following `in most cases you don't need it` may not worth the risk.

Now, if you still want to go for it, the approach would be to create and register a custom decorator that removes the selector from Jobs and ensure (using the before/after) methods that is called after all other decorators that set selectors.

**Comment by iocanel** [comment]
> Option with custom decorator sounds fine. The other options is, whether to include the selector for jobs at first place. As my current fix removes the selector in jobs all together.

The thing with the current fix, is that I can't tell if its going to work or not.

**Comment by The-Huginn** [comment]
It currently removes the `selector` from jobs manifest completely

**Comment by The-Huginn** [comment]
I have only updated the tests, to reflect new changes, i.e. Jobs won't have labels. I kept the original fix. I can try to update it with decorator, just let me know @iocanel

**Comment by The-Huginn** [comment]
Any updates @iocanel ? Currently it contains adjusted tests and completely disabled option for cronjobs to have selector (unless it had defined labels previously). I can try to update it with decorator, it you think that's better.

**Comment by iocanel** [comment]
Will have a look later today

File: integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/KubernetesWithCronJobResourceWithoutSelectorTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+public class KubernetesWithCronJobResourceWithoutSelectorTest {
+
+    static final String APP_NAME = "kubernetes-with-cronjob-resource-without-selector";
+
+    @RegisterExtension
+    static final QuarkusProdModeTest config = new QuarkusProdModeTest()
+            .withApplicationRoot((jar) -> jar.addClasses(GreetingResource.class))
+            .setApplicationName(APP_NAME)
+            .setApplicationVersion("0.1-SNAPSHOT")
+            .withConfigurationResource(APP_NAME + ".properties")
```
Comment (iocanel) [inline integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/KubernetesWithCronJobResourceWithoutSelectorTest.java:30]: For future reference, I find it simpler to have the configuration passed directly tin this file using `overrideKey`.

**Review by iocanel — APPROVED** [review]
LGTM

**Pull request merged**